venter/models.py

```diff
@@ -0,0 +1,127 @@
+from uuid import uuid4
+
+from django.db import models
+from django.utils.timezone import now
+
+STATUS = (
+    ('reported', 'Reported'),
+    ('in_progress', 'In Progress'),
+    ('resolved', 'Resolved'),
+    ('deleted', 'Deleted'),
+)
+
+
+class tag_uris(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    tag_uri = models.CharField(max_length=200)
+
+    class Meta:
+        verbose_name = "Tag URI"
+        verbose_name_plural = "Tag URIs"
+
+    def __str__(self):
+        return self.tag_uri
+
+
+class complaints(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    created_by = models.ForeignKey('users.UserProfile',null=True, blank=True, on_delete=models.CASCADE, related_name="created_by")
+    description = models.TextField(blank=True, null=True)
+    report_date = models.DateTimeField(default=now)
+    status = models.CharField(max_length=30, choices=STATUS, default='reported')
+    latitude = models.FloatField(max_length=8, blank=True, null=True)
+    longitude = models.FloatField(max_length=8, blank=True, null=True)
+    location_description = models.TextField(blank=True, null=True)
+    tags = models.ManyToManyField(tag_uris, related_name='tags', blank=True)
+    users_up_voted = models.ManyToManyField('users.UserProfile', related_name='users_up_voted', blank=True)
+    media = models.ManyToManyField('upload.UploadedImage', related_name='media', blank=True)
+
+    ##
+    class Meta:
+        verbose_name = "Complaint"
+        verbose_name_plural = "Complaints"
+
+    def __str__(self):
+        return self.description
+
+
+class comment(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    time = models.DateTimeField(default=now)
+    text = models.TextField()
+    user = models.ForeignKey('users.UserProfile', on_delete=models.CASCADE)
+    complaint = models.ForeignKey(complaints, on_delete=models.CASCADE, related_name="comments")
+
+    class Meta:
+        verbose_name = "Comment"
+        verbose_name_plural = "Comments"
+
+    def __str__(self):
+        return self.text
+
+
+        # class complaints_liked_users(models.Model):
```

Comment:
No commented code please. See [this](https://softwareengineering.stackexchange.com/questions/190096/can-commented-out-code-be-valuable-documentation) thread.